src/clib/initialize_cloudy_data.c

```diff
@@ -38,17 +53,24 @@ int initialize_cloudy_data(chemistry_data *my_chemistry,
   char parameter_name[MAX_LINE_LENGTH];
 
   // Initialize things needed even if cloudy cooling is not used.
+  initialize_empty_cloudy_data_struct(my_cloudy);
 
-  my_cloudy->grid_parameters = malloc(CLOUDY_MAX_DIMENSION * sizeof(double));
+  my_cloudy->grid_parameters = malloc(CLOUDY_MAX_DIMENSION * sizeof(double *));
   my_cloudy->grid_dimension = malloc(CLOUDY_MAX_DIMENSION * sizeof(long long));
   for (q = 0;q < CLOUDY_MAX_DIMENSION;q++) {
-    my_cloudy->grid_dimension[q] = 0;
+    my_cloudy->grid_dimension[q] = NULL;
+    my_cloudy->grid_parameters[q] = NULL;
   }
 
   // Zero arrays if cloudy cooling not used.
 
   if (read_data == 0) {
     my_cloudy->grid_rank = 0;
+    my_cloudy->cooling_data = NULL;
+    my_cloudy->heating_data = NULL;
+    my_cloudy->cooling_data = NULL;
+    my_cloudy->mmw_data = NULL;
+    my_cloudy->data_size = 0LL;
     return SUCCESS;
   }
```

Comment:
On my first glance, the logic in this snippet seems a little confusing (it seems a little redundant). But, after looking at it more carefully, I think the way you have written it is definitely the best way to do it for the current definition of the `cloudy_data` structure. 

As an aside, I have introduced another PR (#167) that builds on this PR and alters the definition of the `cloudy_data` struct (which seems like a change a little beyond the scope of this PR) in order to simplify the logic in this code snippet.